### PR TITLE
Add document for Testcases

### LINKED ISSUE #192: Add document for Testcases
Add documents (with examples and related references) that explains procedures and necessary guidelines for specifying testcases based on application specific OAS , and implementing testcases in postman

### PR DESCRIPTION
**Pull request by IswaryaaS**
Fixes #192
Adding documents for specifying and implementing testcases in postman

### CODE REVIEW

File: doc/SpecifyingApplications/TestCases/TestCases.md [hunk 1]
```diff
@@ -1,14 +1,101 @@
...
+
+|**Prefix keyword**|**Purpose of test scripts**|
+| :---- | :---- |
+|Reference|Gathering some required information related to the proceeding test script|
+|Expected|Testing expected scenario specific testcases|
+|Dummy|POST/PUT some randomly generated dummy values through into request body|
+|Initial|POST/PUT service/API with Initial values for inversing any potential change in configuration|
+|Clearing|Operations performed for clearing any evident changes in the configuration|
+|sampling|Testing the actual testcase (just for experimental purpose)|
+|Check|Test script that test whether the configuration remains the same as initial data|
```
Comment (PrathibaJee) [inline doc/SpecifyingApplications/TestCases/TestCases.md:29]: If require, can we add 'for gating' , 'for exiting' prefix keyword. 
Please ignore this comment if it is not valid.

File: doc/SpecifyingApplications/TestCases/TestCases.md [hunk 2]
```diff
@@ -1,14 +1,101 @@
...
+
+### ServiceLayer and OamLayer
+
+This folder contains testcases for API enpoints exposed in serviceLayer. It further contains IndividualPart and BasicPart which refers to IndividualServices and BasicServices respectively in OAS specification. Under IndividualPart or BasicPart, testcases for all the services or AI endpoints are listed.
```
Comment (PrathibaJee) [inline doc/SpecifyingApplications/TestCases/TestCases.md:33]: Can this be ,  "This folder contains testcases for API enpoints exposed in serviceLayer **and OamLayer**. "

Comment (PrathibaJee) [inline doc/SpecifyingApplications/TestCases/TestCases.md:33]: "**AI** endpoints are listed." to "**API** endpoints are listed."

File: doc/SpecifyingApplications/TestCases/Testing.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+- Copy the BasicPart from ServiceLayer of ApplicationPattern collection into ServiceLayer of application specific test suite
+- Copy the BasicPart from OamLayer of ApplicationPattern collection into OamLayer of application specific test suite.
+
+Now, The test suite can be tested against the mock server initially for basic corrections before testing against the real server.
+
+*Reference:* [*Mock Server*](../CreatingMock/CreatingMock.md)
+
+## Testing the server
+
```
Comment (PrathibaJee) [inline doc/SpecifyingApplications/TestCases/Testing.md:17]: If possible can we add "customization to do in the load file before running it" ,for eg serverUrl, typeOfTesting , authorizationCode etc., to run the testcase.

**Review by PrathibaJee — CHANGES_REQUESTED** [review]

File: doc/SpecifyingApplications/SpecifyingApplications.md [hunk 4]
```diff
@@ -106,13 +106,19 @@ The CONFIGfile describes the data structure inside the application and on the di
...
 
-The TestCases are a Postman collection of requests for 
-  * describing the business logic, which is to be implemented behind the REST API
-  * supporting testing during implementation
-  * acceptance testing
-  * continuous integration testing
+Test suites are written in postman as collections. A set of testcases are defined for each request that mainly focuses on
+
+- determining if different features/functionality of the service (Rest API) is working fine.
+- that could reveal errors or defects during development of the application
```
[the remark this answers is not in the document]
Comment (IswaryaaS) [reply doc/SpecifyingApplications/SpecifyingApplications.md:112]: Updated as per comment

File: doc/SpecifyingApplications/TestCases/TestCases.md (hunk 2 above)
Comment (IswaryaaS) [reply doc/SpecifyingApplications/TestCases/TestCases.md:33]: updated as per comment

Comment (IswaryaaS) [reply doc/SpecifyingApplications/TestCases/TestCases.md:33]: Updated as per comment

File: doc/SpecifyingApplications/TestCases/TestCases.md (hunk 1 above)
Comment (IswaryaaS) [reply doc/SpecifyingApplications/TestCases/TestCases.md:29]: updated as per comment

File: doc/SpecifyingApplications/TestCases/Testing.md (hunk 3 above)
Comment (IswaryaaS) [reply doc/SpecifyingApplications/TestCases/Testing.md:17]: Updated as per comment

**Review by IswaryaaS — COMMENTED** [review]
Updated the changes requested

**Review by PrathibaJee — DISMISSED** [review]
Thanks for considering and fixing the review comments. 
Approving the merge request.

**Pull request closed without merging**